### PR TITLE
Add configurable default values for group & category on on-the-fly created devices

### PR DESCRIPTION
**Pull request by duke2906**
Hello,

in my opinion a configurable default group and a default category would be usefull when using the optinal functionality to register devices automatically on incoming positions. These are the adjustments i would suggest to implement this enhancement.

### CODE REVIEW

**Comment by tananaev** [comment]
Please fix the errors reported by the build system. Next time you should probably run all the checks locally before pushing any commits.

**Comment by Abyss777** [comment]
Hello,

I think you also need update `groupsByName`  in add/update/removeGroup functions

File: src/org/traccar/BaseProtocolDecoder.java [hunk 1]
```diff
@@ -34,8 +35,17 @@
 
     public long addUnknownDevice(String uniqueId) {
         Device device = new Device();
+        String defaultGroupName = Context.getConfig().getString("database.registerUnknown.defaultGroup", "NOTDEFINED");
```
Comment (tananaev) [inline src/org/traccar/BaseProtocolDecoder.java]: I don't think we need a default value here.

File: src/org/traccar/BaseProtocolDecoder.java [hunk 2]
```diff
@@ -34,8 +35,17 @@
...
         device.setName(uniqueId);
         device.setUniqueId(uniqueId);
+        device.setCategory(Context.getConfig().getString("database.registerUnknown.defaultCategory", "default"));
```
Comment (tananaev) [inline src/org/traccar/BaseProtocolDecoder.java]: Same here. I don't think we need default value.

**Comment by duke2906** [comment]
In removeGroup I implemented a "check" on the old group name to prevent the HashMap from getting bigger and bigger cause of old or doubled group-references when updating this name. Is that correct?

**Comment by tananaev** [comment]
I just realized that we can have group duplicates. I think it's better to use id.

**Comment by duke2906** [comment]
Oh, sorry. I just checked it out...ur right, I am able to create another Group with the same group name. Well ok, than I'll use the id and remove the getGroupByName. Unfortunately the only way to get the GroupID is to query the database, isn't it? Well, that shouldn't be a problem cause this is supposed to be set up once by the administrator, which ofcourse has access to the database...

**Comment by tananaev** [comment]
Admin can always use API to get groups directly, so it shouldn't be a big problem. Plus presumably it needs to be set only once.

File: src/org/traccar/BaseProtocolDecoder.java [hunk 3]
```diff
@@ -34,8 +34,14 @@
 
     public long addUnknownDevice(String uniqueId) {
         Device device = new Device();
+        long defaultGroupId = Context.getConfig().getLong("database.registerUnknown.defaultGroupId");
```
Comment (tananaev) [inline src/org/traccar/BaseProtocolDecoder.java]: Please move this declaration down just before you use the variable.

**Comment by duke2906** [comment]
Whole declaration, or split declaration and initialisation and move
initialisation down right before its used?

Anton Tananaev <notifications@github.com> schrieb am Do., 8. Dez. 2016,
10:13:

**Comment by tananaev** [comment]
Declaration. There is no point in declaring it earlier. It's not JavaScript.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**